pysaql/expression.py

```diff
@@ -1,11 +1,48 @@
 """Contains base Expression class"""
 
 
-class Expression:
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from .util import escape_identifier
+
+
+class Expression(ABC):
     """Base expression class
 
-    This is used as the root class for all expressions in the saql subpackage to
-    establish inheritance. It does not currently implement any common functionality.
+    This is used as the root class for all expressions to establish inheritance and
+    provide common methods.
     """
 
-    pass
+    _alias: Optional[str] = None
+
+    def alias(self, name: str) -> "Expression":
+        """Set the alias name
+
+        Args:
+            name: Alias name
+
+        Returns:
+            self
+
+        """
+        self._alias = name
+        return self
+
+    @abstractmethod
+    def to_string(self) -> str:
+        """Cast the expression to a string"""
+        pass
+
+    def __str__(self) -> str:
+        """Cast the expression to a string, including the alias if set
+
+        Returns:
+            string
+
+        """
+        s = self.to_string()
```

Comment:
Would it make sense to make this `_to_string()` since it seems like users would usually want to use the magic method? Or is there a use case for `to_string()` without the alias logic?